test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java

```diff
@@ -355,4 +355,22 @@ protected boolean forbidPrivateIndexSettings() {
         return true;
     }
 
+
+    /**
+     * waits until all shard initialization is completed.
+     *
+     * @throws IOException
+     */
+    protected void ensureNoInitializingShards() throws IOException {
+        ClusterHealthResponse actionGet = client().admin()
+            .cluster()
+            .health(Requests.clusterHealthRequest("_all").timeout(TimeValue.timeValueSeconds(70)).waitForNoInitializingShards(true))
+            .actionGet();
+
+        assertFalse("timed out waiting for shards to initialize", actionGet.isTimedOut());
+
+        // some tests leave with a non-green state
+        //assertThat(actionGet.getStatus(), equalTo(ClusterHealthStatus.GREEN));
```

Comment:
Why not simply remove these commented lines?

Reply:
I probably should, it feels strange to me, I would expect that the cluster should be healthy, yellow maybe (due to being single node), but red? 

Anyway, no judgement on other tests, maybe its intended.